src/runtime/client-event-manager-test.js

```diff
@@ -0,0 +1,199 @@
+    expect(matchedExpected).to.equal(1);
+    event.eventType = null;
+    expected = createEventErrorMessage('eventType', null);
+    tryIt();
+    expect(errorCount).to.equal(2);
+    expect(matchedExpected).to.equal(2);
+    event.eventType = OTHER_TYPE;
+    tryIt();
+    expect(errorCount).to.equal(2);
+    event.eventType = DEFAULT_TYPE;
+
+    //validate event originator
+    errorCount = 0;
+    matchedExpected = 0;
+    event.eventOriginator = BAD_VALUE;
+    expected = createEventErrorMessage('eventOriginator', BAD_VALUE);
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.eventOriginator = null;
+    expected = createEventErrorMessage('eventOriginator', null);
+    tryIt();
+    expect(errorCount).to.equal(2);
+    expect(matchedExpected).to.equal(2);
+    event.eventOriginator = OTHER_ORIGIN;
+    tryIt();
+    expect(errorCount).to.equal(2);
+    expect(matchedExpected).to.equal(2);
+    event.eventOriginator = DEFAULT_ORIGIN;
+
+    //validate isFromUserAction
+    errorCount = 0;
+    matchedExpected = 0;
+    event.isFromUserAction = BAD_VALUE;
+    expected = createEventErrorMessage('isFromUserAction', BAD_VALUE);
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.isFromUserAction = true;
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.isFromUserAction = false;
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.isFromUserAction = null;
+
+    //validate additionalParameters
+    errorCount = 0;
+    matchedExpected = 0;
+    event.additionalParameters = BAD_VALUE;
+    expected = createEventErrorMessage('additionalParameters', BAD_VALUE);
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.additionalParameters = null;
+    expected = createEventErrorMessage('additionalParameters', null);
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.additionalParameters = {IAmValid: 5};
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+    event.additionalParameters = {};
+
+    //validate null object
+    errorCount = 0;
+    matchedExpected = 0;
+    event = null;
+    expected = 'Event must be a valid object';
+    tryIt();
+    expect(errorCount).to.equal(1);
+    expect(matchedExpected).to.equal(1);
+  });
+
+  it('should be able to listen for events', function*() {
+    const eventMan = new ClientEventManager();
+    let receivedEventsCount = 0;
+    const callback = () => receivedEventsCount++;
+
+    //verify it can listen to 1
+    eventMan.registerEventListener(callback);
+    eventMan.logEvent(DEFAULT_EVENT);
+    yield eventMan.lastAction_;
+    expect(receivedEventsCount).to.equal(1);
+
+    //verify it can listen to 2 at the same time
```

Comment:
do you mean 2 of the same events? we are not sending two events at the same time. This is the second event yes?

Reply:
I mean verify EventManager supports multiple listeners.  It is sending 1 event and there are 2 registered listeners.  Granted its the same listener twice but that shouldn't matter.  It then verifies its listener got called twice after a single event is sent.

Reply:
I did clarify the comment